pkg/cmd/server/kubernetes/master_config.go

```diff
@@ -180,6 +180,25 @@ func BuildKubernetesMasterConfig(options configapi.MasterConfig, requestContextM
 		},
 	}
 
+	if options.DNSConfig != nil {
+		_, dnsPortStr, err := net.SplitHostPort(options.DNSConfig.BindAddress)
+		if err != nil {
+			return nil, fmt.Errorf("unable to parse DNS bind address %s: %v", options.DNSConfig.BindAddress, err)
+		}
+		dnsPort, err := strconv.Atoi(dnsPortStr)
+		if err != nil {
+			return nil, fmt.Errorf("invalid DNS port: %v", err)
+		}
+		m.ExtraServicePorts = append(m.ExtraServicePorts,
```

Comment:
The service is not updated if it already exists. I'm thinking it definitely should be (as endpoints are) and I'm putting together that change. We could simply verify that the ports are present or validate the entire service definition.